### PR TITLE
Revised for API Browser, .NET implementations

### PR DESCRIPTION
**Pull request by rpetrusha**
# Revised for API Browser, .NET implementations

The class library overview is .NET Framework-specific. Since the API Browser provides no overview, it is now more important as a link target, so it should reflect all .NET implementations, not just .NET Framework.

See PR #4336 

[Internal review URL](https://review.docs.microsoft.com/en-us/dotnet/standard/class-library-overview?branch=pr-en-us-4337)

//cc @mairaw @pkulikov

### CODE REVIEW

File: docs/standard/class-library-overview.md
Comment (mairaw) [inline docs/standard/class-library-overview.md]: should we pull out JScript info and add F# instead? /cc @cartermp

File: docs/standard/class-library-overview.md
Comment (mairaw) [inline docs/standard/class-library-overview.md]: the previous two lines has one extra asterisk that is causing formatting issues in the rest of the text. perhaps the asterisk should come after the bold formatting like
`**String**\*` instead of `**String\***`

File: docs/standard/class-library-overview.md
Comment (mairaw) [inline docs/standard/class-library-overview.md]: update ms.date?

**Review by mairaw — CHANGES_REQUESTED** [review]
Changes look great @rpetrusha. Left a couple of comments.

File: docs/standard/class-library-overview.md
Comment (cartermp) [reply docs/standard/class-library-overview.md]: Yep!

File: docs/standard/class-library-overview.md
Comment (pkulikov) [inline docs/standard/class-library-overview.md]: @rpetrusha should we use a relative link to `../../api/index.md` for .NET API Browser? (The same remark refers to the paragraph above.)

Comment (rpetrusha) [reply docs/standard/class-library-overview.md]: The page there appears to be a sub, so I'm not sure that it will render properly, @pkulikov. @mairaw, do you know? Should the link be changed to a relative link in our repo, or should it be left as a link to docs?

File: docs/standard/class-library-overview.md
Comment (rpetrusha) [reply docs/standard/class-library-overview.md]: @mairaw, of course. I always forget about updating the date.

File: docs/standard/class-library-overview.md
Comment (rpetrusha) [reply docs/standard/class-library-overview.md]: I agree, @mairaw. @cartertmp, could you review the table after I merge the revision to make sure that I've correctly described the F# data types? I'm unsure about object. (Is there a need for an object in F#?)

File: docs/standard/class-library-overview.md
Comment (rpetrusha) [reply docs/standard/class-library-overview.md]: The problem here is not only the formatting break, but I'm not sure what * is. (I know it's a pointer, but I'm not sure that String* works in C++.) I've changed the language label to C++/CLI. @ghogen, could you take a look at it for accuracy?

File: docs/standard/class-library-overview.md
Comment (pkulikov) [reply docs/standard/class-library-overview.md]: @rpetrusha ah, OK. I've found the third way that is used currently in the docs. The [raw file](https://raw.githubusercontent.com/dotnet/docs/c6341e749d70800a701288d2dcd2605c7cf4e0f2/docs/framework/get-started/index.md) of **See Also** section of [this page](https://docs.microsoft.com/en-us/dotnet/framework/get-started/) contains the line `[.NET API Browser](/dotnet/api/)`

File: docs/standard/class-library-overview.md
Comment (mairaw) [reply docs/standard/class-library-overview.md]: @mikeblome can you help?

File: docs/standard/class-library-overview.md
Comment (mairaw) [reply docs/standard/class-library-overview.md]: Given that we'll split the repo soon, I think we should keep it as you did.

**Review by mairaw — APPROVED** [review]
LGTM. I wouldn't merge without @cartermp review of the F# types though.

File: docs/standard/class-library-overview.md
Comment (pkulikov) [reply docs/standard/class-library-overview.md]: @mairaw could you please tell more about the split of the repo, or that's too early yet?

Comment (mairaw) [reply docs/standard/class-library-overview.md]: @pkulikov We'll post more details once it's done but we're working to split the conceptual and api docs into two different repos and we'll have a third one for the samples. This should reduce build times, especially for the conceptual docs.

Comment (pkulikov) [reply docs/standard/class-library-overview.md]: @mairaw that's clear, thank you.

**Comment by rpetrusha** [comment]
@cartermp, I've added F# types to the table listing .NET types that are available in individual languages. Could you review it, please? I'm particularly unsure about the System.Object type, though I suspect (quite likely incorrectly) that since F# is a dynamic language there's no need for an explicit object type.

File: docs/standard/class-library-overview.md
Comment (cartermp) [inline docs/standard/class-library-overview.md]: For F#, this should be `obj` instead of `object`

**Review by cartermp — CHANGES_REQUESTED** [review]
One small change, then it looks good!

**Comment by cartermp** [comment]
Just as a note @rpetrusha F# isn't a dynamic language; it's statically and strongly typed, but it uses Hindley-Milner style type inference to eliminate most explicit type annotations.

**Comment by rpetrusha** [comment]
Thanks for the review and the clarification, @cartertmp. I've changed the Object type for F#.

**Comment by rpetrusha** [comment]
@cartertmp, could you take a look at the latest version and approve it if it's OK?

**Review by cartermp — APPROVED** [review]

**Comment by cartermp** [comment]
LGTM!

**Pull request merged**
